### PR TITLE
Imprpve context path

### PR DESCRIPTION
**Pull request by ThibaudAV**
## Issue
n/a

## Description

Following several feedbacks prefer simple add button over complex UX 
image

## Additional context

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/imprpve-context-path/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-amdmmxqirv.chromatic.com)

### CODE REVIEW

**Comment by ytvnr** [comment]
Seeing the screenshot, do you think it would be preferable to have the button at the bottom of the list ? 
What if a user want to add more than 6 entries, it will be obliged to scroll up to click then scroll down to fill the field.
no ?

**Comment by ThibaudAV** [comment]
All the add buttons are on the top right. APIs, Application, others pages, ...
But I must admit I wondered about reversing the 2 buttons. As you also think I will do it

**Comment by ThibaudAV** [comment]
... but not easy

**Comment by ThibaudAV** [comment]
done

**Review by ytvnr — APPROVED** [review]
Looks good to me.
I just have a suggestion in terme of UX, I don't know if it's easy: when clicking on Add context-path, it would be awesome to focus on the new field to be able to type directly
wdyt ?

File: gravitee-apim-console-webui/src/management/api/component/gio-form-listeners/gio-form-listeners-context-path/gio-form-listeners-context-path.component.ts [hunk 1]
```diff
@@ -56,9 +60,13 @@ const PATH_PATTERN_REGEX = new RegExp(/^\/[/.a-zA-Z0-9-_]*$/);
   ],
 })
 export class GioFormListenersContextPathComponent implements OnInit, OnDestroy, ControlValueAccessor, AsyncValidator {
-  public listeners: HttpListenerPath[] = [];
+  public listeners: HttpListenerPath[] = [DEFAULT_LISTNER];
```
Comment (loriepisicchio) [inline gravitee-apim-console-webui/src/management/api/component/gio-form-listeners/gio-form-listeners-context-path/gio-form-listeners-context-path.component.ts]: typo: 
```suggestion
  public listeners: HttpListenerPath[] = [DEFAULT_LISTENER];
```

File: gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.harness.ts [hunk 2]
```diff
@@ -67,10 +67,16 @@ export class Step2Entrypoints2ConfigHarness extends ComponentHarness {
   async fillPaths(...paths: string[]) {
     const formPaths = await this.locatorFor(GioFormListenersContextPathHarness.with())();
 
-    // Add path on last path row
+    // Add fist path on existing ContextPathRow
+    const fistPath = paths.shift();
```
Comment (loriepisicchio) [inline gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.harness.ts]: typo: 
```suggestion
    const firstPath = paths.shift();
```

File: gravitee-apim-console-webui/src/management/api/component/gio-form-listeners/gio-form-listeners-context-path/gio-form-listeners-context-path.component.ts [hunk 3]
```diff
@@ -212,15 +196,6 @@ export class GioFormListenersContextPathComponent implements OnInit, OnDestroy,
     };
   }
 
-  private mustIgnoreLast(listenerFormArrayControls: AbstractControl[]) {
```
Comment (loriepisicchio) [inline gravitee-apim-console-webui/src/management/api/component/gio-form-listeners/gio-form-listeners-context-path/gio-form-listeners-context-path.component.ts:215]: If we delete this, maybe we should make the form invalid until user enters a valid context path after clicking on the Add button. 
For now, the form is considered valid when you add a new row until you touch it.
Screenshot 2023-04-13 at 16 38 27

File: gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.harness.ts (hunk 2 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.harness.ts]: mdr, 🤜

File: gravitee-apim-console-webui/src/management/api/component/gio-form-listeners/gio-form-listeners-context-path/gio-form-listeners-context-path.component.ts (hunk 3 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/component/gio-form-listeners/gio-form-listeners-context-path/gio-form-listeners-context-path.component.ts:215]: corrected but I did not add a test :/ I didn't see how to make it simple

**Review by loriepisicchio — APPROVED** [review]

**Pull request merged**